src/output-json-flow.c

```diff
@@ -346,6 +347,13 @@ static void EveFlowLogJSON(OutputJsonThreadCtx *aft, JsonBuilder *jb, Flow *f)
         jb_close(jb); /* close array */
     }
 
+    if (f->alparser) {
+        uint64_t tx_id = AppLayerParserGetTransactionLogId(f->alparser);
```

Comment:
will this work if no loggers are active?

Reply:
seems like a weird way to get this info, why not use the `AppLayerParserGetTxCnt` func?

Reply:
Indeed.